pkg/parser/uri_test.go

```diff
@@ -0,0 +1,58 @@
+package parser
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+var parseTests = []struct {
+	dsn      string
+	scheme   string
+	path     string
+	rawQuery string
+}{
+	{"file:///some/path/to/file.csv?format=csv", "file", "/some/path/to/file.csv", "format=csv"},
+	{"file:///some/path/to/file.csv", "file", "/some/path/to/file.csv", ""},
+	{"os:stdout?format=csv", "os", "stdout", "format=csv"},
+}
+
+var handledErrors = map[error]bool{
+	ErrEmptyDSN: true,
+}
+
+func TestParseURI(t *testing.T) {
+	for _, test := range parseTests {
+		uri, err := parseURI(test.dsn)
+		// Check that we parse without any unhandled errors
+		if !handledErrors[err] && err != nil {
+			t.Error(
+				"Encountered error when parsing:",
+				err,
+			)
+		}
+		// Check that "scheme", "path" and "query" outputs are correct for non-empty dsns
+		if test.dsn != "" {
+			assert.Equal(t, test.scheme, uri.Scheme, "they should be equal")
+			assert.Equal(t, test.path, uri.Path, "they should be equal")
+			assert.Equal(t, test.rawQuery, uri.RawQuery, "they should be equal")
+		}
+	}
+}
+
+var parseErrorTests = []struct {
+	dsn string
+	err error
+}{
+	{"", ErrEmptyDSN},
```

Comment:
Can you add some more invalid things here? for example `i_am_not_a_uri`, `/`

Reply:
Done.